code/game/objects/items/trader.dm

```diff
@@ -186,4 +186,535 @@
+			verbs += /obj/item/weapon/gun/lawgiver/verb/submit_DNA_sample
+			verbs -= /obj/item/weapon/gun/lawgiver/verb/erase_DNA_sample
+			update_icon()
+		else
+			self_destruct(H)
+
+/obj/item/clothing/head/helmet/donutgiver/proc/self_destruct(mob/user)
+	var/req_access = list(access_security)
+	if(can_access(user.GetAccess(),req_access))
+		say("ERROR: DNA PROFILE DOES NOT MATCH.")
+		return
+	say("UNAUTHORIZED ACCESS DETECTED.")
+	var/datum/organ/external/active_hand = user.get_active_hand_organ()
+	if(active_hand)
+		active_hand.explode()
+	explosion(user, -1, 0, 2)
+	qdel(src)
+
+/obj/item/clothing/head/helmet/donutgiver/Hear(var/datum/speech/speech, var/rendered_speech="")
+	set waitfor = FALSE // speak AFTER the user
+	if(world.timeofday < last_donut+300)
+		return
+	var/dispense_path
+	if(speech.speaker == loc && !speech.frequency && dna_profile)
+		var/mob/living/carbon/human/H = loc
+		if(dna_profile == H.dna.unique_enzymes)
+			if(findtext(speech.message, "standard") || findtext(speech.message, "sprinkle") || findtext(speech.message, "traditional"))
+				dispense_path = /obj/item/weapon/reagent_containers/food/snacks/donut/normal
+				sleep(3)
+				say("SPRINKLE.")
+			else if(findtext(speech.message, "jelly") || findtext(speech.message, "berry") || findtext(speech.message, "juicy"))
+				dispense_path = /obj/item/weapon/reagent_containers/food/snacks/donut/jelly
+				sleep(3)
+				say("JELLY.")
+			else if(findtext(speech.message, "chaos") || findtext(speech.message, "gump") || findtext(speech.message, "two-face"))
+				dispense_path = /obj/item/weapon/reagent_containers/food/snacks/donut/chaos
+				sleep(3)
+				say("CHAOS.")
+			else if(findtext(speech.message, "favorite") || findtext(speech.message, "4Kids") || findtext(speech.message, "rice"))
+				dispense_path = /obj/item/weapon/reagent_containers/food/snacks/riceball
+				sleep(3)
+				say("FAVORITE.")
+		if(dispense_path)
+			var/obj/item/I = new dispense_path(get_turf(src))
+			H.put_in_hands(I)
+			last_donut = world.timeofday
+
+//Security Skirt spritework is coutesy of TG, AGPL license
+/obj/item/clothing/under/securityskirt/elite
+	name = "elite security skirt"
+	desc = "For demonstrating who is in charge."
+	icon_state = "secskirt"
+	item_state = "r_suit"
+	_color = "secskirt"
+	origin_tech = Tc_COMBAT + "=2"
+	armor = list(melee = 10, bullet = 10, laser = 10,energy = 0, bomb = 0, bio = 0, rad = 0)
+	clothing_flags = ONESIZEFITSALL
+	siemens_coefficient = 0.9
+	species_fit = list(GREY_SHAPED) //Unlike normal skirts this is not VOX_SHAPED
+	body_parts_covered = FULL_TORSO|ARMS
+
+/obj/item/clothing/under/securityskirt/elite/equipped(var/mob/user, var/slot)
+	..()
+	processing_objects += src
+
+/obj/item/clothing/under/securityskirt/elite/unequipped(mob/user, var/from_slot = null)
+	processing_objects -= src
+	..()
+
+/obj/item/clothing/under/securityskirt/elite/process()
+	if(prob(1) && ishuman(loc))
```

Comment:
should be `is_wearing_item(src, slot_w_uniform)` or it will do the thing while being held/in a backpack/whatever.

Reply:
Actually, since it only calls `process()` after `equipped()` is called, this is probably fine.